### PR TITLE
secrets: Added input option

### PR DESCRIPTION
**Pull request by jochman**
###  Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/22759

## Description
Added the --input\-i option

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

File: demisto_sdk/commands/secrets/secrets.py [hunk 1]
```diff
@@ -72,8 +79,12 @@ def __init__(self, configuration=Configuration(), is_circle=False, ignore_entrop
     def get_secrets(self, branch_name, is_circle):
         secrets_found = {}
         # make sure not in middle of merge
```
Comment (Itay4) [inline demisto_sdk/commands/secrets/secrets.py:81]: move this comment to the right place

File: demisto_sdk/commands/secrets/secrets.py [hunk 2]
```diff
@@ -72,8 +79,12 @@ def __init__(self, configuration=Configuration(), is_circle=False, ignore_entrop
...
-        if not run_command('git rev-parse -q --verify MERGE_HEAD'):
+        if self.input_path:
+            secrets_file_paths = self.input_path
+        else:
             secrets_file_paths = self.get_all_diff_text_files(branch_name, is_circle)
+        if not run_command('git rev-parse -q --verify MERGE_HEAD') or self.input_path:
+            # If a input path supplied, should not run on git.
```
Comment (Itay4) [inline demisto_sdk/commands/secrets/secrets.py:87]: not sure i understand this comment, lets improve it

File: demisto_sdk/commands/secrets/secrets.py [hunk 3]
```diff
@@ -72,8 +79,12 @@ def __init__(self, configuration=Configuration(), is_circle=False, ignore_entrop
     def get_secrets(self, branch_name, is_circle):
         secrets_found = {}
         # make sure not in middle of merge
-        if not run_command('git rev-parse -q --verify MERGE_HEAD'):
+        if self.input_path:
+            secrets_file_paths = self.input_path
+        else:
             secrets_file_paths = self.get_all_diff_text_files(branch_name, is_circle)
```
Comment (Itay4) [inline demisto_sdk/commands/secrets/secrets.py:85]: if i understand correctly, this was inside the condition because we didnt want to get the diff if we're in the middle of the merge, now its not like that
why did you take it?

File: demisto_sdk/tests/integration_tests/secrets_integration_test.py [hunk 4]
```diff
@@ -133,3 +133,12 @@ def test_integration_secrets_integration_global_whitelist_positive(mocker):
     assert result.exit_code == 0
     assert not result.stderr
     assert "no secrets were found" in result.stdout
+
+
+def test_integration_secrets_integration_positive_with_input_option():
+    integration_secrets_path = join(
+        DEMISTO_SDK_PATH, "tests/test_files/content_repo_example/Packs/FeedAzure/Integrations/FeedAzure/FeedAzure.yml"
+    )
+    chdir(join(DEMISTO_SDK_PATH, "tests", "integration_tests"))
```
Comment (Itay4) [inline demisto_sdk/tests/integration_tests/secrets_integration_test.py:142]: why is this needed, the integration test already runs in this path

File: demisto_sdk/tests/integration_tests/secrets_integration_test.py [hunk 5]
```diff
@@ -133,3 +133,12 @@ def test_integration_secrets_integration_global_whitelist_positive(mocker):
     assert result.exit_code == 0
     assert not result.stderr
     assert "no secrets were found" in result.stdout
+
+
+def test_integration_secrets_integration_positive_with_input_option():
```
Comment (Itay4) [inline demisto_sdk/tests/integration_tests/secrets_integration_test.py:138]: add docstring

Comment (Itay4) [inline demisto_sdk/tests/integration_tests/secrets_integration_test.py:138]: lets also add negative test

File: demisto_sdk/commands/secrets/README.md [hunk 6]
```diff
@@ -21,14 +21,19 @@ demisto-sdk secrets
 This will run the secrets validator on your uncommited files.
 <br/><br/>
 ```
+demisto-sdk secrets -i ./MyRepo/Integration/integration.yml
+```
+This will run the secrets validator on the file located in ./MyRepo/Integration/integration.yml.
```
Comment (Itay4) [inline demisto_sdk/commands/secrets/README.md:26]: provide real example

File: demisto_sdk/commands/common/test_tools.py [hunk 7]
```diff
@@ -0,0 +1,7 @@
+from pathlib import Path
+
+
+def create_temp_file(tmp_path, file_content, filename='file.txt'):
```
Comment (Itay4) [inline demisto_sdk/commands/common/test_tools.py:4]: add type hints

File: demisto_sdk/commands/secrets/secrets.py [hunk 8]
```diff
@@ -72,8 +79,12 @@ def __init__(self, configuration=Configuration(), is_circle=False, ignore_entrop
     def get_secrets(self, branch_name, is_circle):
         secrets_found = {}
         # make sure not in middle of merge
-        if not run_command('git rev-parse -q --verify MERGE_HEAD'):
+        if self.input_path:
+            secrets_file_paths = self.input_path
```
Comment (Itay4) [inline demisto_sdk/commands/secrets/secrets.py:83]:
```suggestion
            secrets_file_paths = [self.input_path]
```

File: demisto_sdk/tests/integration_tests/secrets_integration_test.py (hunk 5 above)
Comment (jochman) [reply demisto_sdk/tests/integration_tests/secrets_integration_test.py:162]: done

File: demisto_sdk/commands/secrets/secrets.py (hunk 8 above)
Comment (jochman) [reply demisto_sdk/commands/secrets/secrets.py:83]: it's already a list (see above)

File: demisto_sdk/commands/common/test_tools.py [hunk 9]
```diff
@@ -1,7 +1,19 @@
 from pathlib import Path
```
Comment (Itay4) [inline demisto_sdk/commands/common/test_tools.py:1]: move this file to `test` dir
